packages/tokamak/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol

```diff
@@ -1,48 +1,73 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.15;
 
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { Predeploys } from "src/libraries/Predeploys.sol";
 import { OptimismPortal } from "src/L1/OptimismPortal.sol";
 import { CrossDomainMessenger } from "src/universal/CrossDomainMessenger.sol";
 import { ISemver } from "src/universal/ISemver.sol";
 import { Constants } from "src/libraries/Constants.sol";
+import { SafeCall } from "src/libraries/SafeCall.sol";
+import { Hashing } from "src/libraries/Hashing.sol";
+import { Encoding } from "src/libraries/Encoding.sol";
+import { Constants } from "src/libraries/Constants.sol";
 
 /// @custom:proxied
 /// @title L1CrossDomainMessenger
 /// @notice The L1CrossDomainMessenger is a message passing interface between L1 and L2 responsible
 ///         for sending and receiving data on the L1 side. Users are encouraged to use this
 ///         interface instead of interacting with lower-level contracts directly.
 contract L1CrossDomainMessenger is CrossDomainMessenger, ISemver {
+    using SafeERC20 for IERC20;
+
     /// @notice Address of the OptimismPortal. The public getter for this
     ///         is legacy and will be removed in the future. Use `portal()` instead.
     /// @custom:network-specific
     /// @custom:legacy
     OptimismPortal public PORTAL;
 
+    /// @notice Address of TON (ERC-20 token)
+    address public l1TONAddress;
```

Comment:
```solidity
 address public immutable l1TONAddress;
```
I think `l1TONAddress` is a knowable constant and we will change the value after deploying this contract. So use `immutable` in this case

What do you think

Reply:
Thank you, Nam
I think you are right

Reply:
As what I checked, other contracts as CrossDomainMessenger, OptimismPortal are not immutable as well. 
So in case of TON, what do you guys think? @rlgns98kr @0x6e616d

Reply:
I don't think there is any way to change the l1tontoken variable. However, unification of variables for each contract seems necessary. What about `nativeTokenAddress?` 😅🙇🏼‍♂️

Reply:
Sure, let me update!

Reply:
I updated!